tsdb/head_append.go

```diff
@@ -350,6 +379,46 @@ func (a *headAppender) AppendExemplar(ref storage.SeriesRef, lset labels.Labels,
 	return storage.SeriesRef(s.ref), nil
 }
 
+func (a *headAppender) AppendMetadata(ref storage.SeriesRef, meta metadata.Metadata) (storage.SeriesRef, error) {
+	// Get Series
+	s := a.head.series.getByID(chunks.HeadSeriesRef(ref))
+
+	// TODO: Do we need to add label set so we can get series by their hash?
+	// if s == nil {
+	// 	s = a.head.series.getByHash(lset.Hash(), lset)
+	// 	if s != nil {
+	// 		ref = storage.SeriesRef(s.ref)
+	// 	}
+	// }
```

Comment:
I think you have to add the label set, yeah. ref, from what I understand, is an optional argument, and the caller might not always have one or save one. lset should be used as a fallback to ensure that the function can't fail if the caller isn't caching ref IDs.